src/librustdoc/json/ids.rs

```diff
@@ -0,0 +1,127 @@
+//! Id handling for rustdoc-json.
+//!
+//! Manages the creation of [`rustdoc_json_types::Id`] and the
+//! fact that these don't corespond exactly to [`DefId`], because
+//! [`rustdoc_json_types::Item`] doesn't corespond exactly to what
+//! other phasises thing of as an "item".
```

Comment:
```suggestion
//! Manages the creation of [`rustdoc_json_types::Id`] and the
//! fact that these don't correspond exactly to [`DefId`], because
//! [`rustdoc_json_types::Item`] doesn't correspond exactly to what
//! other phases think of as an "item".
```